nixos/doc/manual/administration/service-mgmt.xml

```diff
@@ -62,11 +69,63 @@ Jan 07 15:55:57 hagbard systemd[1]: Started PostgreSQL Server.
 # systemctl start postgresql.service
 # systemctl restart postgresql.service
 </screen>
-  These operations are synchronous: they wait until the service has finished
-  starting or stopping (or has failed). Starting a unit will cause the
-  dependencies of that unit to be started as well (if necessary).
- </para>
-<!-- - cgroups: each service and user session is a cgroup
+   These operations are synchronous: they wait until the service has finished
+   starting or stopping (or has failed). Starting a unit will cause the
+   dependencies of that unit to be started as well (if necessary).
+  </para>
+  <!-- TODO: document cgroups, draft:
+   each service and user session is a cgroup
 
-- cgroup resource management -->
+   - cgroup resource management -->
+ </section>
+ <section xml:id="sect-nixos-systemd-nixos">
+  <title>Systemd in NixOS</title>
+  <para>
+   Packages in Nixpkgs sometimes provide Systemd units with them, usually in
+   e.g <literal>#pkg-out#/lib/systemd/</literal>. Putting such a package in
+   <literal>environment.systemPackages</literal> doesn't make the service
+   available to users or the system. Usually system services are defined using
+   NixOS modules written by the community. These services are defined in Nixpkgs'
+   <link xlink:href="https://github.com/NixOS/nixpkgs/tree/master/nixos/modules">
+   <literal>nixos/modules/</literal> directory </link>. We prefer to not rely on
```

Comment:
I don't agree with this statement. We generally prefer to *use* the upstream unit, especially when they contains complex settings like for security hardening. This whould just put more burder on the maintainer and reduce security overall.

Also, pulling in an upstream unit doesn't mean we just hand over control of the service: units can be overriden just fine from the declarative systemd configuration. There are [several]([https://github.com/NixOS/nixpkgs/search?q=use+upstream+unit&type=commits) NixOS module doing this.

Reply:
You are correct. What confused me is that `systemd.packages` enables and starts system service, but not user services on boot. I've documented this after thorough testings.